zcash_test_vectors/transaction_zsa.py

```diff
@@ -0,0 +1,157 @@
+        self.value = rand.u64()
+        asset_digest_bytes = asset_digest(encode_asset_id(ik, asset_desc))
+        self.assetBase = zsa_value_base(asset_digest_bytes)
+        self.rho = Point.rand(rand).extract()
+        self.rseed = rand.b(32)
+
+    def __bytes__(self):
+        ret = b''
+        ret += bytes(self.recipient)
+        ret += struct.pack('<Q', self.value)
+        ret += bytes(self.assetBase)
+        ret += bytes(self.rho)
+        ret += self.rseed
+
+        return ret
+
+
+class TransactionZSA(TransactionBase):
+    def __init__(self, rand, consensus_branch_id, have_orchard_zsa = True, have_burn = True, have_issuance = True):
+
+        # Since burn is part of the OrchardZSA bundle, ensure that there are no burn fields
+        # when there are no OrchardZSA fields
+        assert have_orchard_zsa or not have_burn
+
+        # All the Transparent and Sapling Transaction Fields are initialized in the super (TransactionBase) class.
+        super().__init__(rand, have_orchard_zsa)
+
+        # Common Transaction Fields
+        self.nVersionGroupId = NU7_VERSION_GROUP_ID
+        self.nConsensusBranchId = consensus_branch_id
+
+        # Orchard-ZSA Transaction Fields
+        if have_orchard_zsa:
+            for _ in range(rand.u8() % 5):
+                self.vActionsOrchard.append(OrchardZSAActionDescription(rand))
+            self.flagsOrchard = (self.flagsOrchard & 7) | 4  # Only three flag bits are currently defined, and we set enableZSA to true.
+            if self.is_coinbase():
+                # set enableSpendsOrchard = 0
+                self.flagsOrchard &= 2
+
+        # OrchardZSA Burn Fields
+        self.vAssetBurnOrchardZSA = []
+        if have_burn:
+            for _ in range(rand.u8() % 5):
+                self.vAssetBurnOrchardZSA.append(AssetBurnDescription(rand))
+
+        # ZSA Issuance Fields
+        self.vIssueActions = []
+        if have_issuance:
+            self.ik = IssuanceKeys(rand.b(32)).ik
+            for _ in range(rand.u8() % 5):
+                self.vIssueActions.append(IssueActionDescription(rand, self.ik))
+            self.issueAuthSig = rand.b(64)
+
+    def version_bytes(self):
+        return NU7_TX_VERSION | (1 << 31)
+
+    def orchard_zsa_burn_field_bytes(self):
+        ret = b''
+        ret += write_compact_size(len(self.vAssetBurnOrchardZSA))
+        if len(self.vAssetBurnOrchardZSA) > 0:
+            for desc in self.vAssetBurnOrchardZSA:
+                ret += bytes(desc)
+        return ret
+
+    def issuance_field_bytes(self):
+        ret = b''
+        ret += write_compact_size(len(self.vIssueActions))
+        if len(self.vIssueActions) > 0:
+            for desc in self.vIssueActions:
+                ret += bytes(desc)
+            ret += self.ik
+            ret += bytes(self.issueAuthSig)
+        return ret
+
+    def __bytes__(self):
+        ret = b''
+
+        # Fields that are in TransactionBase: Common, Transparent, Sapling, most Orchard
```

Comment:
`# Common transaction fields` (x2)

Reply:
Similar to comment on line 149. Would it be clear enough?

Reply:
Done, as discussed above.